internal/server/server_test.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/HarlamovBuldog/social-tournament-service/internal/storage"
+)
+
+func TestCreateNewUser_Success(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mock := storage.NewMockService(ctrl)
+	mock.EXPECT().AddUser(gomock.Any(), gomock.Eq("Gennadiy")).Times(1).Return("code_str", nil)
+	s := NewServer(mock)
+	enc, err := json.Marshal(userNameJSON{
+		Name: "Gennadiy",
+	})
+	require.NoError(t, err)
+
+	b := bytes.NewBuffer(enc)
+	req := httptest.NewRequest("GET", "/user", b)
+	w := httptest.NewRecorder()
+	s.createNewUser(w, req)
+
+	actualCode := w.Result().StatusCode
+	assert := assert.New(t)
+	assert.Equal(http.StatusOK, actualCode, "The two http codes should be the same")
```

Comment:
so, why do you want to use assert, but not require?

Reply:
Does it make much difference? I mean from documentation: https://godoc.org/github.com/stretchr/testify

assert | Package assert provides a set of comprehensive testing tools for use with the normal Go testing system.

require | Package require implements the same assertions as the `assert` package but stops test execution when a test fails.

I mean in case when only 1 assert call present there is no difference. But in case u mention, if  actual code doesn't match expected, i guess i still need to decode body to get full picture. If i will use require test will fail before check. That's the point i guess.